test/integration/test_konnect_extension.go

```diff
@@ -53,26 +57,140 @@ func TestKonnectExtensionKonnectGatewayControlPlaneNamespacedRef(t *testing.T) {
+		require.True(t, ok)
+		// TODO: use `WithKonnectIDControlPlaneRef` after KonnectExtension support `SetControlPlaneRef`:
+		// https://github.com/Kong/kubernetes-configuration/issues/328
+		ke.Spec.KonnectControlPlane.ControlPlaneRef = commonv1alpha1.ControlPlaneRef{
+			Type:      commonv1alpha1.ControlPlaneRefKonnectID,
+			KonnectID: lo.ToPtr(cpID),
+		}
+	}
+}
+
+func setKonnectExtesionKonnectNamespacedRefControlPlaneRef(
+	t *testing.T, cp *konnectv1alpha1.KonnectGatewayControlPlane,
+) deploy.ObjOption {
+	return func(obj client.Object) {
+		ke, ok := obj.(*konnectv1alpha1.KonnectExtension)
+		require.True(t, ok)
+		// TODO: use `WithKonnectIDControlPlaneRef` after KonnectExtension support `SetControlPlaneRef`:
+		// https://github.com/Kong/kubernetes-configuration/issues/328
+		ke.Spec.KonnectControlPlane.ControlPlaneRef = commonv1alpha1.ControlPlaneRef{
+			Type: commonv1alpha1.ControlPlaneRefKonnectNamespacedRef,
+			KonnectNamespacedRef: &commonv1alpha1.KonnectNamespacedRef{
+				Name:      cp.Name,
+				Namespace: cp.Namespace,
+			},
+		}
+	}
+}
+
+func setKonnectExtensionDPCertSecretRef(t *testing.T, s *corev1.Secret) deploy.ObjOption {
+	return func(obj client.Object) {
+		ke, ok := obj.(*konnectv1alpha1.KonnectExtension)
+		require.True(t, ok)
+		ke.Spec.DataPlaneClientAuth = &konnectv1alpha1.DataPlaneClientAuth{
+			CertificateSecret: konnectv1alpha1.CertificateSecret{
+				Provisioning: lo.ToPtr(konnectv1alpha1.ManualSecretProvisioning),
+				CertificateSecretRef: &konnectv1alpha1.SecretRef{
+					Name: s.Name,
+				},
+			},
+		}
+	}
+}
+
+func checkKonnectExtensionConditions(t *assert.CollectT, ke *konnectv1alpha1.KonnectExtension) (bool, string) {
+	err := GetClients().MgrClient.Get(GetCtx(), types.NamespacedName{Name: ke.Name, Namespace: ke.Namespace}, ke)
+	require.NoError(t, err)
+
+	checkConditionTypes := []consts.ConditionType{
+		konnectv1alpha1.ControlPlaneRefValidConditionType,
+		konnectv1alpha1.DataPlaneCertificateProvisionedConditionType,
+		konnectv1alpha1.KonnectExtensionReadyConditionType,
+	}
+	return helpers.CheckAllConditionsTrue(ke, checkConditionTypes)
+}
+
+func checkKonnectExtensionStatus(
+	ke *konnectv1alpha1.KonnectExtension,
+	expectedKonnectCPID string,
+	expectedDPCertificateSecretName string,
+) func(t *assert.CollectT) {
+	return func(t *assert.CollectT) {
+		err := GetClients().MgrClient.Get(GetCtx(), types.NamespacedName{Name: ke.Name, Namespace: ke.Namespace}, ke)
+		require.NoError(t, err)
+		// Check Konnect control plane ID
+		assert.NotNil(t, ke.Status.Konnect, "status.konnect should be present")
+		assert.Equal(t, expectedKonnectCPID, ke.Status.Konnect.ControlPlaneID, "Konnect control plane ID should be set in status")
+		// Check dataplane client auth
+		assert.NotNil(t, ke.Status.DataPlaneClientAuth, "status.dataPlaneClientAuth should be present")
+		assert.NotNil(t, ke.Status.DataPlaneClientAuth.CertificateSecretRef, "status.dataPlaneClientAuth.certiifcateSecretRef should be present")
```

Comment:
```suggestion
		require.NotNil(t, ke.Status.DataPlaneClientAuth, "status.dataPlaneClientAuth should be present")
		require.NotNil(t, ke.Status.DataPlaneClientAuth.CertificateSecretRef, "status.dataPlaneClientAuth.certiifcateSecretRef should be present")
```

Reply:
Ditto here for `status.dataplaneClientAuth`.